### PR TITLE
Update dotnet extension versions

### LINKED ISSUE #1576: Update .NET worker item templates for latest extension versions
The templates use fairly old versions of the worker extensions. These need to be updated to pull in the latest versions and validated.

As a concrete example, we don't currently pull in (transitively) the Service Bus versions that tolerate the "https" scheme being applied to the `fullyQualifiedNamespace`.

All .NET worker item templates should be tackled by this work.

### PR DESCRIPTION
**Pull request by liliankasem**
Update all extension versions to the latest for csharp and fsharp projects.

resolves #1576

### CODE REVIEW

File: Functions.Templates/Templates/RabbitMQTrigger-CSharp/.template.config/template.json [hunk 1]
```diff
@@ -51,7 +51,8 @@
             "ManualInstructions": [],
             "args": {
                 "referenceType": "package",
-            "reference": "Microsoft.Azure.WebJobs.Extensions.RabbitMQ", "version": "1.1.0",
+                "reference": "Microsoft.Azure.WebJobs.Extensions.RabbitMQ",
+                "version": "2.0.3",
```
Comment (surgupta-msft) [inline Functions.Templates/Templates/RabbitMQTrigger-CSharp/.template.config/template.json:55]: Are we comfortable with the major version update here?

File: Functions.Templates/Templates/MySqlOutputBinding-CSharp/.template.config/template.json [hunk 2]
```diff
@@ -47,7 +47,8 @@
           "ManualInstructions": [],
           "args": {
             "referenceType": "package",
-            "reference": "Microsoft.Azure.WebJobs.Extensions.MySql", "version": "0.0.000",
+            "reference": "Microsoft.Azure.WebJobs.Extensions.MySql",
+            "version": "1.0.2-preview",
```
Comment (surgupta-msft) [inline Functions.Templates/Templates/MySqlOutputBinding-CSharp/.template.config/template.json:51]: major version change?

File: Functions.Templates/Templates/MySqlOutputBinding-CSharp-Isolated/.template.config/template.json [hunk 3]
```diff
@@ -47,7 +47,8 @@
           "ManualInstructions": [],
           "args": {
             "referenceType": "package",
-            "reference": "Microsoft.Azure.Functions.Worker.Extensions.MySql", "version": "0.0.000",
+            "reference": "Microsoft.Azure.Functions.Worker.Extensions.MySql",
+            "version": "1.0.2-preview",
```
Comment (surgupta-msft) [inline Functions.Templates/Templates/MySqlOutputBinding-CSharp-Isolated/.template.config/template.json:51]: major version change?

Comment (liliankasem) [reply Functions.Templates/Templates/MySqlOutputBinding-CSharp-Isolated/.template.config/template.json:51]: I think `0.0.000` is just a placeholder that was never updated with the version of "MySql" package when it became available. I validated the template with this change and didn't see any issues. (same for comment above for the csharp proj)

Comment (liliankasem) [reply Functions.Templates/Templates/MySqlOutputBinding-CSharp-Isolated/.template.config/template.json:51]: Also noticed that with this set to v0, the extension is never loaded so the project doesn't even work. Adding a real version here also fixes this issue.

File: Functions.Templates/Templates/RabbitMQTrigger-CSharp/.template.config/template.json (hunk 1 above)
Comment (liliankasem) [reply Functions.Templates/Templates/RabbitMQTrigger-CSharp/.template.config/template.json:55]: Validating nothing is broken here, should know soon!

Comment (surgupta-msft) [reply Functions.Templates/Templates/RabbitMQTrigger-CSharp/.template.config/template.json:55]: Can we leave this one unchanged? Any drawbacks of doing that?

Comment (liliankasem) [reply Functions.Templates/Templates/RabbitMQTrigger-CSharp/.template.config/template.json:55]: No issues with the template, works as expected

**Review by mattchenderson — APPROVED** [review]

**Review by surgupta-msft — APPROVED** [review]

**Pull request merged**